packages/custom-elements/src/CustomElementInternals.js

```diff
@@ -317,7 +286,12 @@ export default class CustomElementInternals {
    */
   upgradeReaction(element) {
     try {
-      this._upgradeAnElement(element);
+      const definition = this._lookupACustomElementDefinition(
```

Comment:
Seems like this is being setup to support scoped registries. So just thinking ahead, do you think we'd need to call `getRootNode` on the element here?

Reply:
(from offline) Exactly,`_lookupACustomElementDefinition` would likely end up using `element.getRootNode()` to get the nearest document or shadow root and use the registry there, or fall back to the registry associated with `element.getRootNode({composed: true})`.